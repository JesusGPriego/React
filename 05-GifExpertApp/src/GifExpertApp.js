import React from "react";
// import PropTypes from "prop-types";
import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";
//rafcp: commando to fast creating a component.

/**
 * It is the main component on the app. it displays two different components: <br >
 * an AddCategory component and undefined number of GifGrid.
 * The first is an input text and the others are 'galleries' of images.
 */
export const GifExpertApp = ({defaultCategories = []}) => {
  //useState hook. for every new category, a new Grid item will be created
  const [categories, setCategories] = useState(defaultCategories);

 

  return (
    <>
      <h2>GifExpertApp</h2>
      {/* comunicating components  */}
      <AddCategory setCategories={setCategories} />
      <hr />
      <ul>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ul>
    </>
  );
};

// GifExpertApp.propTypes = {};
