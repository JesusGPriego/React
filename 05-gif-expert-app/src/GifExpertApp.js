import React from "react";
import PropTypes from "prop-types";
//rafcp
export const GifExpertApp = (props) => {
  return (
    <>
      <h2>GifExpertApp</h2>
      <hr />
    </>
  );
};

GifExpertApp.propTypes = {};
