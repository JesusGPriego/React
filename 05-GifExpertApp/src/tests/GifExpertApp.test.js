import { shallow } from "enzyme";
import { GifExpertApp } from '../GifExpertApp';



describe('Testing GifExpertApp', () => {
    test('should match snapshot', () => {

       const wrapper =  shallow(<GifExpertApp/>)
       expect(wrapper).toMatchSnapshot();

    });

    test('should show "categories.length" GifGrid components', () => {
       
        const categories = ['one punch man', 'Dragon ball z'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    });
    
    
})
