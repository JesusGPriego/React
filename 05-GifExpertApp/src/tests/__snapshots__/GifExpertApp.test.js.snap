// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifExpertApp should match snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;

exports[`Testing GifExpertApp should show "categories.length" GifGrid components 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="one punch man"
      key="one punch man"
    />
    <GifGrid
      category="Dragon ball z"
      key="Dragon ball z"
    />
  </ul>
</Fragment>
`;
